### PR TITLE
FlashLightVisualizer refactor + low power handheld light light radius fix

### LINKED ISSUE #11610: Floodlight light radius is not properly restored
## Description
Whenever the current power cell reaches a level that causes the floodlight's light radius to change, that radius will be retained even if you give it a fully charged power cell later on.

**Reproduction**
1. Have full power cell
2. Add to floodlight then turn it on
3. Wait until it goes under 40% when it starts flashing
4. Remove power cell then recharge it
5. Add power cell back
6. :(

**Video**
https://user-images.githubusercontent.com/14862618/193137725-e55c38fd-81db-43b8-afd1-0eb7ca2da713.mp4

### PR DESCRIPTION
**Pull request by Carbonhell**
## About the PR 
Fixes #11610 
Requires https://github.com/space-wizards/RobustToolbox/pull/3345
I also deleted the FlashLightVisualizer (it used the obsolete AppearanceVisualizer methods instead of events) and the LanternVisualizer (same reason, but nothing was using it as well).

**Changelog**

:cl:
- fix: Low power lighting radius animations for lanterns, floodlights and flashlights now properly restore the original light radius when going back to full power.

### CODE REVIEW

**Comment by mirrorcult** [comment]
isnt there light animation behaviour stuff that already exists for these, i thought thats what they used

**Comment by Carbonhell** [comment]
> isnt there light animation behaviour stuff that already exists for these, i thought thats what they used

I'm certain the visualizer is what was used before. I also tried to look for similar animations across the codebase or specific implementation in the clientside handheldlight component but I've found nothing similar, do you have anything more specific I can look for?

**Comment by mirrorcult** [comment]
what im thinking of is `LightBehaviourComponent` which does handle that

```yaml
    - type: LightBehaviour
      behaviours:
        - !type:RandomizeBehaviour # immediately make it bright and flickery
          id: turn_on
          interpolate: Nearest
          minDuration: 0.02
          maxDuration: 0.06
          startValue: 6.0
          endValue: 9.0
          property: Energy
          isLooped: true
        - !type:FadeBehaviour # have the radius start small and get larger as it starts to burn
          id: turn_on
          interpolate: Linear
          maxDuration: 8.0
          startValue: 1.0
          endValue: 6.0
          property: Radius
        - !type:RandomizeBehaviour # weaker flicker as it fades out
          id: fade_out
          interpolate: Nearest
          minDuration: 0.02
          maxDuration: 0.06
          startValue: 4.0
          endValue: 8.0
          property: Energy
          isLooped: true
        - !type:FadeBehaviour # fade out radius as it burns out
          id: fade_out
          interpolate: Linear
          maxDuration: 4.0
          startValue: 6.0
          endValue: 1.0
          property: Radius
```

**Comment by mirrorcult** [comment]
so these should probably use that if at all possible

**Comment by Carbonhell** [comment]
> so these should probably use that if at all possible

Thank you, I've refactored the code to use light behaviours and I think it's a lot more flexible now. I had to do some small changes to LightBehaviourComponent. I've tested all the possible battery state changes and the animations and the transitions both work correctly.

**Comment by Carbonhell** [comment]
The engine update required for this PR has been integrated in the content repo with the engine version 0.55.5.0 at commit ece9bf372abfbcd78c83eec75024a47de6a96949 so this PR should be good to go after a review.

File: Content.Shared/Light/Component/HandheldLightComponent.cs [hunk 1]
```diff
@@ -45,6 +45,20 @@ public sealed class HandheldLightComponent : Robust.Shared.GameObjects.Component
 
         public const int StatusLevels = 6;
 
+        /// <summary>
+        /// Specify the ID of the light behaviour to use when the state of the light is Dying
+        /// </summary>
+        [ViewVariables]
+        [DataField("blinkingBehaviourID")]
```
Comment (DrSmugleaf) [inline Content.Shared/Light/Component/HandheldLightComponent.cs:52]:
```suggestion
        [DataField("blinkingBehaviourId")]
```

File: Content.Shared/Light/Component/HandheldLightComponent.cs [hunk 2]
```diff
@@ -45,6 +45,20 @@ public sealed class HandheldLightComponent : Robust.Shared.GameObjects.Component
...
+        public string BlinkingBehaviourId { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Specify the ID of the light behaviour to use when the state of the light is LowPower
+        /// </summary>
+        [ViewVariables]
+        [DataField("radiatingBehaviourID")]
```
Comment (DrSmugleaf) [inline Content.Shared/Light/Component/HandheldLightComponent.cs:59]:
```suggestion
        [DataField("radiatingBehaviourId")]
```

File: Resources/Prototypes/Entities/Objects/Misc/fluff_lights.yml [hunk 3]
```diff
@@ -6,6 +6,29 @@
   components:
   - type: HandheldLight
     addPrefix: true
+    blinkingBehaviourID: blinking
+    radiatingBehaviourID: radiating
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Entities/Objects/Misc/fluff_lights.yml:10]:
```suggestion
    blinkingBehaviourId: blinking
    radiatingBehaviourId: radiating
```

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**